sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/Batchers.java

```diff
@@ -155,6 +160,75 @@ public final void batch(Labels labelSet, Aggregator aggregator, boolean unmapped
               aggregation.getDescriptorType(descriptor.getType(), descriptor.getValueType()),
               points));
     }
+
+    @Override
+    public boolean generatesDeltas() {
+      return delta;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+
+      AllLabels allLabels = (AllLabels) o;
+
+      if (startEpochNanos != allLabels.startEpochNanos) {
+        return false;
+      }
+      if (delta != allLabels.delta) {
+        return false;
+      }
+      if (descriptor != null
+          ? !descriptor.equals(allLabels.descriptor)
+          : allLabels.descriptor != null) {
+        return false;
+      }
+      if (aggregation != null
+          ? !aggregation.equals(allLabels.aggregation)
+          : allLabels.aggregation != null) {
+        return false;
+      }
+      if (resource != null ? !resource.equals(allLabels.resource) : allLabels.resource != null) {
+        return false;
+      }
+      if (instrumentationLibraryInfo != null
+          ? !instrumentationLibraryInfo.equals(allLabels.instrumentationLibraryInfo)
+          : allLabels.instrumentationLibraryInfo != null) {
+        return false;
+      }
+      if (clock != null ? !clock.equals(allLabels.clock) : allLabels.clock != null) {
+        return false;
+      }
+      if (aggregatorFactory != null
+          ? !aggregatorFactory.equals(allLabels.aggregatorFactory)
+          : allLabels.aggregatorFactory != null) {
+        return false;
+      }
+      return aggregatorMap != null
+          ? aggregatorMap.equals(allLabels.aggregatorMap)
+          : allLabels.aggregatorMap == null;
+    }
+
+    @Override
+    public int hashCode() {
+      int result = descriptor != null ? descriptor.hashCode() : 0;
```

Comment:
Think you can use `Objects.hashCode` instead of your own null checks

Reply:
I just had IDEA generate this code. Hate to mess with generated code in general. 
We should document what we want to do with generated equals/hashcode/toString implementations, so we don't have to discuss it. :)